multi-repo-action/org-workflow-add.go

```diff
@@ -0,0 +1,106 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+
+	"github.com/google/go-github/v42/github"
+	"golang.org/x/oauth2"
+)
+
+// *****SET THESE PARAMETERS*****
+const ORG_NAME string = "ORGANIZATION NAME"
+const PAT string = "PERSONAL ACCESS TOKEN"
+
+var REPO_LIST = []string{} // OPTIONAL, LEAVE EMPTY TO PROCESS ALL REPOS UNDER ORG
+
+// Adds the OpenSSF Scorecard workflow to all repositores under the given organization.
+func main() {
+	// Get github user client.
+	context := context.Background()
+	tokenService := oauth2.StaticTokenSource(
+		&oauth2.Token{AccessToken: PAT},
+	)
+	tokenClient := oauth2.NewClient(context, tokenService)
+	client := github.NewClient(tokenClient)
+
+	// If not provided, get all repositories under organization.
+	if len(REPO_LIST) == 0 {
+		lops := &github.RepositoryListByOrgOptions{Type: "all"}
+		repos, _, err := client.Repositories.ListByOrg(context, ORG_NAME, lops)
+		err_check(err, "List Org Repos Error")
+
+		// Convert to list of repository names.
+		for _, repo := range repos {
+			REPO_LIST = append(REPO_LIST, *repo.Name)
+		}
+	}
+
+	// Get yml file into byte array.
+	workflowContent, err := ioutil.ReadFile("scorecards-analysis.yml")
```

Comment:
It may be useful to automatically generate the default branch and randomize the schedule, as done in https://github.com/actions/starter-workflows/blob/main/code-scanning/scorecards.yml#L6-L8 (next PR)